pkg/server/ptypes/runner.go

```diff
@@ -4,13 +4,30 @@ import (
 	validation "github.com/go-ozzo/ozzo-validation/v4"
 	"github.com/imdario/mergo"
 	"github.com/mitchellh/go-testing-interface"
+	"github.com/mitchellh/hashstructure/v2"
 	"github.com/stretchr/testify/require"
 
 	"github.com/hashicorp/waypoint/internal/pkg/validationext"
 	serverpkg "github.com/hashicorp/waypoint/pkg/server"
 	pb "github.com/hashicorp/waypoint/pkg/server/gen"
 )
 
+// RunnerLabelHash calculates a unique hash for the set of labels on the
+// runner. This generates a consistent hash value for an empty set of labels.
+// The result is never 0.
+func RunnerLabelHash(v map[string]string) (uint64, error) {
+	if v == nil {
+		v = map[string]string{}
+	}
+
+	// We always set this special key so that even empty label sets have
+	// a non-zero hash value. This MUST NEVER BE CHANGED otherwise the
+	// hash values for all previously issued tokens will be invalidated.
+	v["waypoint.hashicorp.com/runner-hash"] = "1"
```

Comment:
At the risk of sounding pedantic, but considering the imporatnance stressed in this code comment, is it worth adding a test to ensure `RunnerLabelHash(nil)` always returns the expected value? (`9642272480295632742` according to running it locally).

Reply:
Good idea.